packages/angular_devkit/schematics/tools/ajv-option-transform_spec.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+// tslint:disable:no-any
+import 'rxjs/add/operator/mergeMap';
+import { AjvSchemaRegistry } from './ajv-option-transform';
+
+
+describe('AjvSchemaRegistry', () => {
+  it('works asynchronously', done => {
+    const registry  = new AjvSchemaRegistry();
+    const data: any = {};  // tslint:disable:no-any
+
+    registry
+      .compile({
+        properties: {
+          bool: { type: 'boolean' },
+          str: { type: 'string', default: 'someString' },
+          obj: {
+            properties: {
+              num: { type: 'number' },
+              other: { type: 'number', default: 0 },
+            },
+          },
+          tslint: {
+            $ref: 'http://json.schemastore.org/tslint#',
+          },
+        },
+      })
+      .mergeMap(validator => validator(data))
+      .map(result => {
+        expect(result.success).toBe(true);
+        expect(data.obj.num).toBeUndefined();
+        expect(data.tslint).not.toBeUndefined();
+      })
+      .subscribe(done, done.fail);
+    });
+
+  it('works synchronously', done => {
```

Comment:
There should be a way to fail synchronously if it's not sync. Otherwise you can't really know if it is sync or async and need to always prepare for it to be async.

Reply:
You can add the check, but this is used internally by ourselves so I don't see the need.

Reply:
Added a comment with this information.